class/defaults.yml

```diff
@@ -18,6 +18,41 @@ parameters:
 
     monitoring:
       enabled: true
+      alerts:
+        KyvernoAverageAdmissionLatencyHigh:
+          enabled: true
+          rule:
+            annotations:
+              description: Kyverno average admission latency for {{$labels.resouce_kind}} reached 5 seconds.
+              message: Kyverno average admission latency for {{$labels.resouce_kind}} reached 5 seconds in the last 10 minutes.
+              runbook_url: https://hub.syn.tools/kyverno/runbooks/KyvernoAverageAdmissionLatencyHigh.html
+              summary: Kyverno admission is slowing down.
+            expr: |
+              (
+                sum by (job,resource_kind) (rate(kyverno_admission_review_duration_seconds_sum[5m]))
+              /
+                sum by (job,resource_kind) (rate(kyverno_admission_review_duration_seconds_count[5m]))
+              ) > 5
+            for: 10m
+            labels:
+              severity: warning
+        KyvernoAdmissionLatencyReachedWebhookTimeout:
+          enabled: true
+          rule:
+            annotations:
+              description: Kyverno admission latency for {{$labels.resouce_kind}}/{{$labels.resource_namespace}} higher than Kubernetes webhook timeout (10s).
+              message: Kyverno admission latency higher than Kubernetes webhook timeout (10s).
+              runbook_url: https://hub.syn.tools/kyverno/runbooks/KyvernoAdmissionLatencyReachedWebhookTimeout.html
+              summary: Kyverno admission too slow.
+            expr: |
+              (
+                sum by(resource_kind, resource_namespace) (increase(kyverno_admission_review_duration_seconds_bucket{le="+Inf"}[5m]))
+              -
+                sum by(resource_kind, resource_namespace) (increase(kyverno_admission_review_duration_seconds_bucket{le="10"}[5m]))
+              ) > 0
+            for: 5m
```

Comment:
I have two questions for this rule:

1. Does kyverno actually report admission latency over 10s or does it also have a timeout there? 
2. Couldn't this rule potentially miss a single slow admission? We take the increase over 5m. That means for a single slow admission this expression will be true for exactly 5m, right? So with the `for: 5m` we might miss this one or at the very least the alert will be immediately closed, or am I completely missing how these rules work? :sweat_smile:

Reply:
> 1. Does kyverno actually report admission latency over 10s or does it also have a timeout there?

It does from my testing.

> 2. I think you are right. This is racy at worst and does nothing at best.